server/proxy/pf_channels.c

```diff
@@ -46,13 +47,27 @@ void pf_OnChannelConnectedEventHandler(void* data,
 {
 	pClientContext* pc = (pClientContext*) data;
 	pServerContext* ps = pc->pdata->ps;
-
 	WLog_INFO(TAG, "Channel connected: %s", e->name);
 
 	if (strcmp(e->name, RDPEI_DVC_CHANNEL_NAME) == 0)
 	{
 		pc->rdpei = (RdpeiClientContext*) e->pInterface;
 	}
+	else if (strcmp(e->name, RAIL_SVC_CHANNEL_NAME) == 0)
+	{
+		RailClientContext* client;
+		RailServerContext* server;
+		client = (RailClientContext*) e->pInterface;
+		pc->rail = client;
+		server = ps->rail;
+		pf_rail_pipeline_init(client, server, pc->pdata);
```

Comment:
do not register client callbacks until rail server is open, it might create a segfault (client might receive a message from remote server while proxy's rail server is not initialized, then it would crash).